api/v1/backup_types.go

```diff
@@ -328,10 +328,14 @@ func (backupStatus *BackupStatus) SetAsCompleted() {
 // SetAsStarted marks a certain backup as started
 func (backupStatus *BackupStatus) SetAsStarted(targetPod *corev1.Pod, method BackupMethod) {
 	backupStatus.Phase = BackupPhaseStarted
-	backupStatus.InstanceID = &InstanceID{
-		PodName:     targetPod.Name,
-		ContainerID: targetPod.Status.ContainerStatuses[0].ContainerID,
+
+	if len(targetPod.Status.ContainerStatuses) > 0 {
+		backupStatus.InstanceID = &InstanceID{
+			PodName:     targetPod.Name,
+			ContainerID: targetPod.Status.ContainerStatuses[0].ContainerID,
+		}
 	}
+
```

Comment:
I think it's incorrect to continue if the ContainedID is not detected correctly.

Reply:
In order to apply this change, should I change SetAsStarted signature to check an error when it is invoked?
The error is returned if containerID is not detected properly.

Reply:
I would not change SetAsStarted's signature.
But I would check there are ContainerStatuses before we call SetAsStarted.
Also, are you doing the changes @gabriele-wolfox , or am I?